services/hh/processor.go

```diff
@@ -144,8 +157,25 @@ func (p *Processor) Process() error {
 	p.mu.RLock()
 	defer p.mu.RUnlock()
 
-	res := make(chan error, len(p.queues))
-	for nodeID, q := range p.queues {
+	// Only process queues for active nodes.
+	var activeQueues map[uint64]*queue
+	if p.MetaProvider == nil {
```

Comment:
I wanted to make the Provider optional, so if you want all the complexity, you add it. Otherwise the Provider will just handle the nodes it got. It also makes testing cleaner. I don't feel strongly about this.